mendix/web-widgets
# fix(tabbing): tab on filter dropdown losses focus

### Pull request type
Bug Fix

---

### Description
 Pressing tab on the last item or shift & tab on the first item of the filter selector list would cause the widget to lose focus (document.activeElement went back to the body).

| Tab | Shift & Tab |
| --- | --- |
|  ![Screenshot 2023-09-20 at 9 31 37 AM](https://github.com/mendix/web-widgets/assets/47681665/dfa97f68-8c17-49a6-8738-cb47f15981db)  | ![Screenshot 2023-09-20 at 9 31 30 AM](https://github.com/mendix/web-widgets/assets/47681665/84c4b0dc-7258-4579-925d-082f5925907b)   |

Changed on KeyDown for the FilterSelector component to check this scenario and set focus back to the filter selector button with a 100ms timeout. 

Testing: 
1. Recompile the Text Filter, Number Filter, and Date Filter widgets
2. Use all of the widgets in Data Grid 2 with "Adjustable by User" set to Yes
3. Run the project and click on one of the filter's button  
![Screenshot 2023-09-20 at 9 38 53 AM](https://github.com/mendix/web-widgets/assets/47681665/75a84a46-bc73-4ba6-8167-ce90c516b432)  
4. Pressing tab on the last item, shift & tab on the first item, or the escape key on any item should close the dropdown and re-focus the button from step 3 allowing you to tab into the text box.
